ckanext/canada/plugins.py

```diff
@@ -564,17 +576,29 @@ def get_auth_functions(self):
     # IXloader
 
     def can_upload(self, resource_id):
+        """
+        Only uploaded resources are allowed to be xloadered, or white listed sources.
+        """
 
         # check if file is uploded
         try:
             res = p.toolkit.get_action(u'resource_show')({'ignore_auth': True},
                                                          {'id': resource_id})
 
-            if res.get('url_type', None) != 'upload':
+            if res.get('url_type') != 'upload' and res.get('url_type') != '' and res.get('url_type') is not None:
                 log.error(
-                    'Only uploaded resources can be added to the Data Store.')
+                    'Only uploaded resources and white listed sources can be added to the Data Store.')
                 return False
 
+            if not res.get('url_type'):
+                allowed_domains = p.toolkit.config.get('ckanext.canada.datastore_source_domain_allow_list', [])
+                url = res.get('url')
+                url_parts = urlsplit(url)
+                if url_parts.netloc not in allowed_domains:
+                    log.error(
+                        'Only uploaded resources and white listed sources can be added to the Data Store.')
```

Comment:
is it appropriate to log an error when a resource can't be uploaded? This method is here to check if it can be uploaded so "no" seems like a normal case not an error, right?